### PR TITLE
Add draft CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by dancingbug**
Right now this is only a draft.
Related issue: #60 

#### TODO/Decisions:
* [ ] Details setting up testing using Docker locally (can someone else give me a suggestion for this?)
* [ ] Brasil folks: What do you want to say here?
* [ ] Decide whether to make the 'sub-official' Telegram official and put it here and/or README.md
* [ ] Decide whether to give commit message formatting guidelines?
* [ ] Decide if documentation about testing should go somewhere else? 
* [ ] Details about manual Firebird db setup and setting up an alias using aliases.conf
* [ ] Platform specific notes
* [ ] Other additions/suggestions?
* [ ] Links to/from other docs
* [ ] Inter-section links
* [ ] Table of contents

### CODE REVIEW

**Comment by afsmi** [comment]
> Decide whether to make the 'sub-official' Telegram official and put it here and/or README.md

I think some web service may be better where you do not have to give a phone number. I use Telegram for some things but maybe it is a little privacy concern for some people ( I feel this way a little bit )

I will look at [gitter.im](https://gitter.im).  it's not obvious to me if their interface is english-only.

any other suggestions here?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+* Updates to out-of-date information, and small changes to existing files can be submitted directly as a PR. 
+
+### In all cases
+* Please read the section about Github Workflow. It is important that submitted PRs pass our automated tests and are correctly formatted.
+* If you are not sure about something, please check existing documentation before creating a PR or issue.  Since this crate is still fairly young, it is considerate, but not required, to also search for information that has already been discussed in old PRs and issues.
+* Note that this repository holds the source for several crates. If proposed changes are very drastic, it may be appropriate to implement it in a separate crate and/or put it behind a feature gate.
+* You may be asked to write or modify some tests after submitting an issue or PR. We will help with this where we can, if needed. See the section on testing for more information.
+
+### A template for commit messages
+(TODO)
```
Comment (fernandobatels) [inline CONTRIBUTING.md:21]: A good suggestion for this template: https://github.com/torvalds/subsurface-for-dirk/blob/a48494d2fbed58c751e9b7e8fbff88582f9b2d02/README#L88

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+You can (and should) run these tests for your configuration using the command `cargo test -- --test-threads 1` 
+
+See the section on testing for more information.
+
+## Testing
+### Setting up a test database `test.fdb`
+#### Using the docker image
+  TODO
+#### Manually
+  * To set up a Firebird database for testing you will need to install the Firebird server. This process may vary by platform. (TODO: notes for different platforms/server versions)
```
Comment (fernandobatels) [inline CONTRIBUTING.md:37]: We can refer to the [official documentation](https://www.firebirdsql.org/file/documentation/html/en/firebirddocs/qsg3/firebird-3-quickstartguide.html#qsg3-installing-server)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  * With a version of the server installed, you may run these commands to create a database: (TODO)
```
Comment (fernandobatels) [inline CONTRIBUTING.md:38]:
```suggestion
  * With a version of the server installed:
    - Using the isql, run `CREATE DATABASE 'localhost:test.fdb';`
    - Or using other tools, create a 'test.fdb' database
```

**Comment by fernandobatels** [comment]
> > Decide whether to make the 'sub-official' Telegram official and put it here and/or README.md

I use the telegram because the activity of rust group there. We can use the telegram for now and make a gitter account too. 
 
> I think some web service may be better where you do not have to give a phone number. I use Telegram for some things but maybe it is a little privacy concern for some people ( I feel this way a little bit )

You can [hide your phone number](https://social.techjunkie.com/hide-phone-number-telegram/) in the telegran.

> any other suggestions here?

We can use a mailgroup into google groups, like the [firebird-support](https://groups.google.com/g/firebird-support)

**Comment by fernandobatels** [comment]
>     * Other additions/suggestions?

We can create the README.md for each crete, referring his own docs. Today we only have a link to the main crate docs

**Comment by afsmi** [comment]
voila https://gitter.im/rsfbclient/rsfbclient (gitter is a bit funny about naming communities vs rooms. maybe it's the same on other sites)

@fernandobatels and @jairinhohw  
I hope you do not mind that I send you invitations. If you don't want to use gitter.im at all, that's OK

**Review by fernandobatels — APPROVED** [review]

**Comment by fernandobatels** [comment]
Continuing on #107

**Pull request closed without merging**